.github/workflows/node_modules.yml

```diff
@@ -0,0 +1,258 @@
+name: npm install runner
+
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  start:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 5
+    permissions:
+      pull-requests: write
+    env:
+      GIT_DIR: git-dir.git
+      BASE_SHA: ${{ github.sha }}
+
+    steps:
+      - name: Interpret comment
+        id: interpret
+        uses: actions/github-script@v4
+        with:
+          result-encoding: string
+          script: |
+            // Make sure we're being spoken to
+            const { action, issue, comment: { body, id: comment_id } } = context.payload;
+            if (!(action === 'created' && body === '/npm install' && 'pull_request' in issue)) {
+                // HACK: That's not really a "fail", but it's the easiest
+                // and safest way to make sure everything else gets skipped
+                core.setFailed('Not our command');
+                return;
+            }
+
+            // Check the permissions of the sender
+            const permission_response = await github.repos.getCollaboratorPermissionLevel({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                username: context.payload.sender.login
+            });
+            const permission = permission_response.data.permission;
+            if (permission !== 'admin' && permission !== 'write') {
+                await github.reactions.createForIssueComment({
+                    owner: context.repo.owner,
+                    repo: context.repo.repo,
+                    comment_id: context.payload.comment.id,
+                    content: 'confused'
+                });
+                await github.issues.createComment({
```

Comment:
Any chance this could edit the existing comment? it seems all too easy to create an infinite loop here, as this workflow triggers itself right now.

(As discussed, I'd *really* like this to be a task in the description, or a label instead, which would be structurally much simpler -- with a label you can even skip the whole permission check)

Reply:
I thought about the label thing too and even preferred it (and for the same reason) but it means that the entire workflow would end up running any time any label is set, including reporting a full set of irrelevant statuses to the PR.

Reply:
ooh, because it's `on: pull_request_target` type `labelled`? I was hoping we could use the [`on: issues`](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#issues) trigger with type "labeled", but now I remember that this literally only works on issues. (I fell into that trap before already -- this is where workflow triggers and webhook events differ).

So let's ignore this for now.